sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveExternalCatalogSuite.scala

```diff
@@ -71,4 +72,19 @@ class HiveExternalCatalogSuite extends ExternalCatalogSuite {
     assert(!rawTable.properties.contains(HiveExternalCatalog.DATASOURCE_PROVIDER))
     assert(DDLUtils.isHiveTable(externalCatalog.getTable("db1", "hive_tbl")))
   }
+
+  test("alter table schema should save case-sensitive schema to properties") {
```

Comment:
as long as the returned table schema is corrected, we don't need to care about these special table properties. We already have a lot of tests to guarantee that table schema always matches the special table properties.